src/index.js

```diff
@@ -77,6 +83,8 @@ class HCaptcha extends React.Component {
       const hcaptcha = this._hcaptcha;
       const captchaId = this.captchaId;
 
+      this._cancelPendingExecute('react-component-unmounted');
```

Comment:
Maybe it should be after the isReady command? Maybe under hcaptcha.remove incase the core hCaptcha code already rejects the promise.